pkg/frontend/kubeactions/gatherer/gatherer.go

```diff
@@ -0,0 +1,58 @@
+package gatherer
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"io"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/env"
+	"github.com/Azure/ARO-RP/pkg/util/portforward"
+)
+
+// Interface contains functions called from within MustGather which need to be mockable
+type Interface interface {
+	WaitForPodRunning(ctx context.Context, k8scli kubernetes.Interface, pod *corev1.Pod) error
+	CopyMustGatherLogs(ctx context.Context, oc *api.OpenShiftCluster, pod *corev1.Pod, w io.Writer) error
+}
```

Comment:
AFAICS the only reason for this package and interface is "it's hard to mock and unit test these two functions".
With WaitForPodRunning, take the pain.
With CopyMustGatherLogs, change portforward.ExecStdout so that **it** is an interface, and we can deal with unit testing that some other time.
Then this entire package melts away.

Reply:
@jim-minter I think this is ready for another look